tls/s2n_recv.c

```diff
@@ -131,6 +132,9 @@ ssize_t s2n_recv(struct s2n_connection *conn, void *buf, ssize_t size, s2n_block
             if (r == -2) {
                 conn->closed = 1;
                 *blocked = S2N_NOT_BLOCKED;
+                if (s2n_is_caching_enabled(conn->config) && conn->session_id_len) {
+                    conn->config->cache_delete(conn->config->cache_delete_data, conn->session_id, conn->session_id_len);
+                }
```

Comment:
I suppose we're being conservative here and assuming that if the client closes on us while we're still trying to talk to them, something bad happened and we should expire the entry.